### DISCUSSION TITLE
cumulating Alpine.data in an x-data element with spread operator "..." don't trigger all init()

### DISCUSSION THREAD

**Discussion by Ronan-Lenor** [alpinejs/alpine#3025] [7. Bugs (archived)] [answered]
Hello,

when i use a script that look like this (simplified) :
```html
<script>
document.addEventListener('alpine:init', () => {
    Alpine.data('A', () => ({
        init()  { alert(111);  },
    }));
    Alpine.data('B', () => ({
        init()  { alert(222);  },
    }));
});
</script>
<div x-data="{ ...A() , ...B() }">test</div>
```
i only get the `alert(222)` to work.
it's like if the `init()` from `A` have been ereased by `B` 🤷‍♂️

how come this happen?
Alpine isn't supposed to seperate functions in their attributed object properties `Alpine.data('storeTheFunctionInThisKey', () =>...` so it can execute each `init()` separatly ???

**Comment by SimoTod** [comment, marked as answer]
You are merging A and B, only one of the 2 init is kept since you can't have a duplicate key in an object.
It's javascript, not Alpine

**Reply by SimoTod** [reply]
To clarify, the reason is this line: `x-data="{ ...A() , ...B() }"`

**Reply by Ronan-Lenor** [reply]
oh thanks.
damn, for a random reason i thought it was merging the `Alpine.data`, not the value (here the function itself) of each properties.
